invenio_drafts_resources/services/records/service.py

```diff
@@ -375,6 +376,15 @@ def publish(self, identity, id_, uow=None, expand=False):
         """
         # Get the draft
         draft = self.draft_cls.pid.resolve(id_, registered_only=False)
+
+        # If config is True and there are no communities selected
+        # Then, check for permissions to upload without community
+        if (
+            current_app.config["RDM_RECORD_ALWAYS_IN_COMMUNITY"]
```

Comment:
the configuration variable is declared in rdm-records, therefore this module knows nothing about it if they are not installed together. 
In addition, drafts resources does not know anything about invenio-communities, therefore this check should be done in a place that is aware of both drafts and communities